tests/unit/test_config.py

```diff
@@ -232,7 +246,33 @@ def test_legacy_variables_override_if_given(self, default_ip):
             edge_port_http,
         ) = config.populate_legacy_edge_configuration(environment)
 
-        assert gateway_listen == f"{default_ip}:4566"
+        assert gateway_listen == [HostAndPort(host=default_ip, port=4566)]
         assert edge_bind_host == "192.168.0.1"
         assert edge_port == 10101
         assert edge_port_http == 20202
+
+    # generate random hostnames and ports from within the following criteria:
+    # - hostnames: a-zA-Z0-9-.
+    # - ports: 0-65535
+    @given(text(alphabet=SAMPLER, min_size=1), integers(min_value=0, max_value=2**16 - 1))
```

Comment:
I think it has a great use in the project, since we do a lot of parsing. I don't have any immediate plans, but I see potential in the future. Do you think it's not worth the extra package?

The single test here gains a lot of coverage since we test many inputs and ensure the output is sensible.

Reply:
Okay, sounds like a good idea to keep it. It's not too big, seems to be well maintained and prominent, and can be quite useful for future test.